pom.xml

```diff
@@ -281,6 +314,12 @@ and
       <artifactId>groovy-console</artifactId>
       <version>${groovy.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.sisu</groupId>
+      <artifactId>org.eclipse.sisu.inject</artifactId>
+      <version>${maven-sisu.version}</version>
+      <classifier>no_asm</classifier>
```

Comment:
Since ASM is a transitive dependency of this repository anyway, it seems consistent to use it throughout rather than using it in some places but not others. Plus this makes the code (very slightly) easier to maintain by avoiding the need for a custom classifier:

```suggestion
```

I tested this locally and tests seemed to work fine.

Reply:
I think I understand this now: this seems to be avoiding a shaded copy of ASM. That is better for us, since we already have ASM in our tree and have no need to shade it. If my understanding is correct, I think we should keep the `no_asm` classifier and resolve this conversation.